### ISSUE TITLE
Support the use of menus in card actions

### ISSUE THREAD

**Issue by matsjoyce** [aforemny/material-components-web-elm#121]
In the material design docs (https://material.io/components/cards#actions), cards can have overflow menus that consist of a icon button and a menu. Currently the `Card` API only supports the addition of normal icon buttons (or rather `Card.icon`), and so adding a icon button with a menu is only possible if you ignore the `actions` parameter and create your own version, which is annoying due to the number of CSS classes that have to be applied.

As a solution, buttons (both `Button` and `IconButton`) could support a new config function named `setMenu`:

```elm
let
    moreMenu = 
        Menu.menu
            (Menu.config
                |> Menu.setOpen True
                |> Menu.setOnClose MenuClosed
            )
            [ -- ...
            ]
in
Card.card Card.config
    { blocks =
        [ -- ...
        ]
    , actions =
        Just <|
            Card.actions
                { buttons = []
                , icons =
                    [ Card.icon
                        ( IconButton.config
                            |> IconButton.setMenu moreMenu
                        )
                        "more_vert"
                    ]
                }
    }
```
The signature could be one of:
```elm
setMenu : Just (Html msg) -> Config msg -> Config msg

setMenu : List (Html msg) -> Config msg -> Config msg
```

The button creation function would set the `Menu.surfaceAnchor` as required. I'm happy to provide an implementation of this, if you think the API is suitable and consistent.

**Comment by aforemny** [comment]
Hi @matsjoyce, thank you for raising this issue. I think it would be great if we had support for overflow menus in Cards! 

While I really like the idea of supporting menus in Buttons and Icon Buttons, I would like to contrast that solution to a solution just within Cards. How could an API for that look like? Does the recent support of `IconButton.custom` bring us anything in that regard? Is it already feasible to use `IconButton.custom` to have Menus in Cards?

Regarding your suggestions, I have the following questions:

(1) Suppose there is an `IconButton.setMenu` configuration option. How would the DOM of `IconButton.iconButton (IconButton.config |> IconButton.setMenu …`) look like? I fear it would have to be `List (Html m)` instead of `Html m`. (Even if that is the case, we could consider adding a `IconButton.iconButtonWithMenu : … -> List (Html m)` primitive.)

(2) Currently menus are opened and closed explicitly through Elm messages. I am assuming that if a Menu is paired with say an Icon Button, the opening and closing should happen automatically. How do we treat event handlers `IconButton.setOnClick` and configuration options `Menu.setOpen`, since I imagine those could still be specified?

**Comment by matsjoyce** [comment]
The reason I proposed adding it to buttons is that it's the smallest change I can think of to achieve this in a nice way. As a side effect, it makes buttons with menus easier generally, which is probably where most menus are used. The alternative is a card specific api like `Card.iconWithMenu`, which doesn't seem better to me, or allowing any HTML elements into a card's action block, which would be complicated due to the extra css classes that need to be applied.

With `IconButton.custom` you could add the menu like this:
```elm
IconButton.iconButton 
    ( IconButton.config
        |> IconButton.setAttributes [ Menu.surfaceAnchor ]
        |> IconButton.setOnClick (ToggleMenu True)
    )
    (IconButton.customIcon Html.div []
        [ Icon.icon [] "more_vert"
        , Menu.menu
            (Menu.config
                |> Menu.setOpen menuOpen
                |> Menu.setOnClose (ToggleMenu False)
            )
            [ List.list
                (List.config |> List.setWrapFocus True)
                (ListItem.listItem ListItem.config
                    [ text "Menu item" ]
                )
                [ ListItem.listItem ListItem.config
                    [ text "Menu item" ]
                ]
            ]
        ]
    )
    
-- vs

IconButton.iconButton
    ( IconButton.config
        |> IconButton.setOnClick (ToggleMenu True)
        |> IconButton.setMenu
            ( Just <| Menu.menu
                (Menu.config
                    |> Menu.setOpen menuOpen
                    |> Menu.setOnClose (ToggleMenu False)
                )
                [ List.list
                    (List.config |> List.setWrapFocus True)
                    (ListItem.listItem ListItem.config
                        [ text "Menu item" ]
                    )
                    [ ListItem.listItem ListItem.config
                        [ text "Menu item" ]
                    ]
                ]
            )
    )
    "more_vert" 
```
So that would work (and I think this works for buttons as well). It just seems a bit awkward and non-intuitive.

(1) For a button, it would look like:

```elm
button variant ((Config { additionalAttributes, touch, href, menu }) as config_) label =
    let
        wrapTouch node = -- ...
        wrapMenu node =
          case menu of
            Nothing ->
              node
            Just m ->
              Html.div [ Menu.surfaceAnchor ]
                [ node
                , m
                ]
    in
    Html.node "mdc-button"
      -- ...
      |> wrapTouch
      |> wrapMenu
```

(2) I was just thinking about doing it manually, since users may want to disable the menu or open it programatically. In that approach, users would use `Menu.setOpen` and `Button.setOnClick` as normal. Making the opening automatic would probably require some JS code, and seems less "Elmish" IMO, since no message would be produced.

**Comment by aforemny** [comment]
Hi @matsjoyce, thank you for your reply and apologies for my late feedback. Would you be still interested in implementing this? I would merge this change. :)

**Comment by matsjoyce** [comment]
Sure, I'll put a PR together. Don't worry about late feedback, nothing on my end is being held back by this.

**Comment by aforemny** [comment]
Merged in 43ae4f1a09bd3c01d5c4750545034266f6c76bc7.
